execution/executor-types/src/lib.rs

```diff
@@ -255,11 +257,11 @@ impl ExecutedTrees {
 
     pub fn version(&self) -> Option<Version> {
         let num_elements = self.txn_accumulator().num_leaves() as u64;
-        if num_elements > 0 {
-            Some(num_elements - 1)
-        } else {
-            None
+        let version: (u64, bool) = num_elements.overflowing_sub(1);
+        if version.1 {
+            return None;
         }
+        Some(version.0)
```

Comment:
this is not great ^ rewrite as:

```rust
num_elements.checked_sub(1)
```

which will return Some or None directly

Reply:
oh, that's right , much easier